### PR TITLE
Use Path in the compiler

### PR DESCRIPTION
**Pull request by andylokandy**
1. Move the changes in https://github.com/idris-lang/Idris2/pull/123 from `libs/contrib/System/Path.idr` to `src/Utils/Path.idr` due to the namespace resolution issue (https://github.com/idris-lang/Idris2/issues/140).

2. Refactor the compiler code to manipulate paths using `Utils.Path` instead of String appending.

I think the `Utils.Path` could eventually be moved into `base`, just like other utils modules in the compiler.

### CODE REVIEW

File: src/Utils/Path.idr [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+dirSeparator = if isWindows then '\\' else '/'
+
+||| The character that separates multiple paths.
+export
+pathSeparator : Char
+pathSeparator = if isWindows then ';' else ':'
+
+||| Windows' path prefixes of path component.
+public export
+data Volumn
```
Comment (melted) [inline src/Utils/Path.idr:32]: Should that be Volume?

File: src/Utils/Path.idr [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+import Text.Lexer
+import Text.Parser
+import Text.Quantity
+
+infixr 5 </>
+infixr 7 <.>
+
+
+isWindows : Bool
+isWindows = os `elem` ["windows", "mingw32", "cygwin32"]
```
Comment (melted) [inline src/Utils/Path.idr:18]: We should probably move this to System.Info, it exists in Core.Options too.

Comment (andylokandy) [reply src/Utils/Path.idr:18]: Yes! I'm still working in the process. Maybe I should change the title a bit...

**Comment by melted** [comment]
You can press "Convert to draft" in the sidebar if you don't want anyone to merge it accidentally.

**Comment by andylokandy** [comment]
> You can press "Convert to draft" in the sidebar if you don't want anyone to merge it accidentally.

But I can't run the checks in a draft.

**Comment by melted** [comment]
No need of a PR for the checks, they should run in your repo too, check the Actions tab.

**Comment by andylokandy** [comment]
@melted Thanks!

**Comment by andylokandy** [comment]
@melted PTAL

**Comment by melted** [comment]
It looks good! I'll wait with merging for a bit to see if there are any other comments.

**Pull request merged**